### ISSUE TITLE
ElasticMeterRegistry reports NaN values to ES causing errors

### ISSUE THREAD

**Issue by tkp1n** [micrometer-metrics/micrometer#497]
The `ElasticMeterRegistry` may report NaN for doubles which causes an error on Elasticsearch.

The culprit (most likely at least):
https://github.com/micrometer-metrics/micrometer/blob/3ec44d958736b0e4c6979479cbc2982fd491384b/implementations/micrometer-registry-elastic/src/main/java/io/micrometer/elastic/ElasticMeterRegistry.java#L310

The erorr as reported by Elasticsearch:
```json
"index": {
    "_index": "metrics-2018.03.15",
    "_type": "doc",
    "_id": "AWIqq3txcLB5uE05seul",
    "status": 400,
    "error": {
        "type": "mapper_parsing_exception",
        "reason": "failed to parse",
        "caused_by": {
            "type": "i_o_exception",
            "reason": "Non-standard token 'NaN': enable JsonParser.Feature.ALLOW_NON_NUMERIC_NUMBERS to allow\n at [Source: org.elasticsearch.common.bytes.BytesReference$MarkSupportingStreamInputWrapper@32e12734; line: 1, column: 98]"
        }
    }
}
```

Possible solutions:
- Skip the meter if numeric value is not a number
- Report 0.0 instaed of NaN

/cc @jkschneider

**Comment by jkschneider** [comment]
Gauge/TimeGauge are the only things that can report NaN. I guess we'll skip reporting them if they get to that value.

**Comment by tkp1n** [comment]
I just retested your fixes to #497, #498, and #499.
So far everything is looking good! Thanks a lot
